src/dbnode/persist/fs/seek_manager.go

```diff
@@ -228,8 +259,8 @@ func (m *seekerManager) CacheShardIndices(shards []uint32) error {
 		byTime := m.seekersByTime(shard)
 
 		byTime.Lock()
-		// Track accessed to precache in open/close loop
-		byTime.accessed = true
+		// Mark seekers to be precached in open/close loop
+		byTime.state = seekersByTimeCached
```

Comment:
Mark state cached in Test() function on line 299 as well. If caller calls Test() after closing a shard then the cached seekers should stay open. Also add a unit-test to verify this behavior.

Reply:
Any thoughts on this?